core/common/src/main/java/alluxio/uri/StandardURI.java

```diff
@@ -162,7 +162,8 @@ public int compareTo(URI other) {
       if ((compare = mPort - other.getPort()) != 0) {
         return compare;
       }
-    } else if ((compare = mAuthority.compareTo(other.getAuthority())) != 0) {
+    } else if ((compare = URIUtils.compare(mAuthority == null ? null : mAuthority.toString(),
```

Comment:
Another way to address this sort of problem is to add a new `NoAuthority` class which extends `Authority` and represents not having any authority, so the `toString()` would just return empty string.

Reply:
NoAuthority will simplify our codes here but will add more complexity if we use Authority type Constructors in AlluxioURI.
Authority authority == null && NoAuthority authority both means we do not have authority.

Reply:
The purpose of using `NoAuthority` is to have more graceful handling of the `null` case, so that the authority would never be set to `null`. Right now we only allow `authority` to be `null` so that we can handle to no-authority case. If we introduce `NoAuthority`, we can guarantee that `authority` is never `null`, so we can call `toString()` without needed a `null` check.